### PR TITLE
Explains how to import ts files instead of js files

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Since we migrated to TypeScript, modules devs importing hardlinks may have some problems without the ts-loader, here is a doc to explain it. It also add some explanations about the typescript implementation.

### CODE REVIEW

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+menuTitle: Import core js files
+weight: 4
+aliases:
+- 1.7/modules/templating/
+---
+
+# How to import core js files
+{{< minver v="1.7.8" title="true">}}
+
+Most of times, you want to use the [global JavaScript components][global JavaScript components] object but in some cases, you may wants to import a component using a static path such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.
```
Comment (atomiix) [inline src/content/1.7/modules/concepts/templating/import-js.md:12]:
```suggestion
Most of the time, you want to use the [global JavaScript components][global JavaScript components] object but in some cases, you may want to import a component using a static path such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+```javascript
+// Incorrect
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.js'
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.ts'
+
+// Correct
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search'
+```
+
+As you're importing this, we consider that you're using a bundler such as Webpack or Parcel. If the file you're trying to import is a TypeScript file, you'll need to add the `TypeScript loader` to you're project and a `tsconfig` file.
```
Comment (atomiix) [inline src/content/1.7/modules/concepts/templating/import-js.md:25]:
```suggestion
As you're importing this, we consider that you're using a bundler such as Webpack or Parcel. If the file you're trying to import is a TypeScript file, you'll need to add the `TypeScript loader` to your project and a `tsconfig` file.
```

**Review by PierreRambaud — APPROVED** [review]

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+title: How to import core js files
```
Comment (matks) [inline src/content/1.7/modules/concepts/templating/import-js.md:2]:
```suggestion
title: How to import Core JavaScript files 
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+---
+title: How to import core js files 
+menuTitle: Import core js files
```
Comment (matks) [inline src/content/1.7/modules/concepts/templating/import-js.md:3]: Maybe a bit more formal? 😉 

```suggestion
menuTitle: Import Core JavaScript files
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+aliases:
+- 1.7/modules/templating/
+---
+
+# How to import core js files
+{{< minver v="1.7.8" title="true">}}
+
+Most of the time, you want to use the [global JavaScript components][global JavaScript components] object but in some cases, you may want to import a component using a static path such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.
+
+**We do not recommend to do this**, but in case you absolutly need it be careful to import the file without any extensions:
```
Comment (matks) [inline src/content/1.7/modules/concepts/templating/import-js.md:14]:
```suggestion
**We do not recommend to do this**, but in case you absolutely need it, be careful to import the file without any extensions:
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+Example of a TypeScript loader:
+```javascript
+{
+  test: /\.ts?$/,
+  use: 'ts-loader',
+  exclude: /node_modules/,
+},
+```
+
+Example of a TypeScript config file:
```
Comment (matks) [inline src/content/1.7/modules/concepts/templating/import-js.md:36]:
```suggestion
Example of a TypeScript configuration file:
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+```javascript
+// Incorrect
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.js'
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.ts'
+
+// Correct
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search'
+```
+
+As you're importing this, we consider that you're using a bundler such as Webpack or Parcel. If the file you're trying to import is a TypeScript file, you'll need to add the `TypeScript loader` to your project and a `tsconfig` file.
```
Comment (matks) [inline src/content/1.7/modules/concepts/templating/import-js.md:25]:
```suggestion
If you need to carry out the above import, it is likely you use a bundler such as Webpack or Parcel. If the file you are trying to import is a TypeScript file, you need to add the `TypeScript loader` to your project and a `tsconfig` file.
```

**Review by matks — COMMENTED** [review]
Some suggestions for more formal style, and one small typo

File: src/content/1.7/development/coding-standards/_index.md [hunk 8]
```diff
@@ -118,6 +118,19 @@ If you need to deprecate services, you can use the `deprecated` key for the serv
...
 Javascript files MUST follow the [Airbnb Javascript style guide](https://github.com/airbnb/javascript).
 
+Starting on {{< minver v="1.7.8" >}}, you can run the linter like this:
+
+```bash
+npm run lint-fix
+```
+
+## TypeScript
+Starting on {{< minver v="1.7.9" >}}, most of `admin-dev/themes/new-theme/js` files have been migrated to TypeScript. This provide a better maintainability by using types on most of our classes and functions.
```
Comment (eternoendless) [inline src/content/1.7/development/coding-standards/_index.md:128]: This part should be done in a different PR

File: src/content/1.7/modules/concepts/templating/import-js.md (hunk 4 above)
Comment (eternoendless) [reply src/content/1.7/modules/concepts/templating/import-js.md:3]: Maybe it's too long for the menu?

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+---
+title: How to import core js files 
+menuTitle: Import core js files
+weight: 4
+aliases:
+- 1.7/modules/templating/
+---
+
+# How to import core js files
```
Comment (eternoendless) [inline src/content/1.7/modules/concepts/templating/import-js.md:9]:
```suggestion
# How to import Core JavaScript files
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
+---
+title: How to import core js files 
+menuTitle: Import core js files
+weight: 4
+aliases:
+- 1.7/modules/templating/
```
Comment (eternoendless) [inline src/content/1.7/modules/concepts/templating/import-js.md:6]: Where does this alias come from? I don't see any page there currently.

File: src/content/1.7/modules/concepts/templating/import-js.md (hunk 4 above)
Comment (NeOMakinG) [reply src/content/1.7/modules/concepts/templating/import-js.md]: Looks fine tbh 

image

File: src/content/1.7/modules/concepts/templating/import-js.md (hunk 10 above)
Comment (NeOMakinG) [reply src/content/1.7/modules/concepts/templating/import-js.md:6]: src/content/1.7/modules/concepts/templating/admin-views.md 

Comes from this file, I removed it in import-js, do you want me to remove it from admin-views?

File: src/content/1.7/development/coding-standards/_index.md [hunk 11]
```diff
@@ -118,6 +118,11 @@ If you need to deprecate services, you can use the `deprecated` key for the serv
 
 Javascript files MUST follow the [Airbnb Javascript style guide](https://github.com/airbnb/javascript).
 
+Starting on {{< minver v="1.7.8" >}}, you can run the linter like this:
```
Comment (eternoendless) [inline src/content/1.7/development/coding-standards/_index.md:121]:
```suggestion
Starting on {{< minver v="1.7.8" >}}, you can run the linter to help you comply with these coding standards:
```

File: src/content/1.7/modules/concepts/templating/import-js.md (hunk 10 above)
Comment (eternoendless) [reply src/content/1.7/modules/concepts/templating/import-js.md]: Yep, that's a mistake, existing pages shouldn't be aliased.

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
+---
+title: How to import Core JavaScript files 
+menuTitle: Import Core JavaScript files
+weight: 4
+---
+
+# How to import Core JavaScript files
+{{< minver v="1.7.8" title="true">}}
+
+Most of the time, you want to use the [global JavaScript components][global JavaScript components] object but in some cases, you may want to import a component using a static path such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.
```
Comment (eternoendless) [inline src/content/1.7/modules/concepts/templating/import-js.md:10]:
```suggestion
Most of the time, you will want to use the [global JavaScript components][global JavaScript components] object. However, in some cases, you may want to import a component using a static path, such as `admin-dev/themes/new-theme/js/components/auto-complete-search.ts`.
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+**We do not recommend to do this**, but in case you absolutely need it, be careful to import the file without any extensions:
```
Comment (eternoendless) [inline src/content/1.7/modules/concepts/templating/import-js.md:12]:
```suggestion
**We don't recommend doing this.** You can still do it if you absolutely need it, but in that case, be careful to import the file without any extensions:
```

File: src/content/1.7/modules/concepts/templating/import-js.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+```javascript
+// Incorrect
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.js'
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search.ts'
+
+// Correct
+import AutoCompletSearch from '../../../../admin-dev/themes/new-theme/js/components/auto-complete-search'
+```
+
+If you need to carry out the above import, it is likely you use a bundler such as Webpack or Parcel. If the file you are trying to import is a TypeScript file, you need to add the TypeScript loader to your project and a `tsconfig` file.
```
Comment (eternoendless) [inline src/content/1.7/modules/concepts/templating/import-js.md:23]:
```suggestion
If you need to carry out the above import, it is likely you are using a bundler such as Webpack or Parcel. If the file you are trying to import is a TypeScript file, you need to add the TypeScript loader to your project and a `tsconfig` file.
```

**Review by eternoendless — APPROVED** [review]

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @NeOMakinG
